### PR TITLE
refactor code about `query -> plan` for subqueries

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

Closes #.

# Rationale for this change

After #4484 #4141 , we remove the alias in `query_to_plan()`, so we can remove it.

What's more, we can find just `subquery_to_plan` need to pass `outer_schema`.

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/sql/src/planner.rs [hunk 1]
```diff
@@ -452,12 +448,9 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                     ))));
                 }
                 // create logical plan & pass backreferencing CTEs
-                let logical_plan = self.query_to_plan_with_alias(
-                    *cte.query,
-                    None,
-                    &mut planner_context.clone(),
-                    outer_query_schema,
-                )?;
```
Comment (jackwener) [inline datafusion/sql/src/planner.rs:460]: just `subquery -> plan` need to pass outer_query_schema.

File: datafusion/sql/src/planner.rs [hunk 2]
```diff
@@ -416,19 +416,15 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
...
-    pub fn query_to_plan_with_alias(
+    /// Generate a logic plan from an SQL query.
+    /// It's implementation of `subquery_to_plan` and `query_to_plan`.
+    /// It shouldn't be invoked directly.
+    fn query_to_plan_with_schema(
         &self,
         query: Query,
-        alias: Option<String>,
         planner_context: &mut PlannerContext,
         outer_query_schema: Option<&DFSchema>,
```
Comment (alamb) [inline datafusion/sql/src/planner.rs:429]: 👍

File: datafusion/sql/src/planner.rs [hunk 3]
```diff
@@ -416,19 +416,15 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
...
-            None,
-            planner_context,
-            Some(outer_query_schema),
-        )
+        self.query_to_plan_with_schema(query, planner_context, Some(outer_query_schema))
     }
 
-    /// Generate a logic plan from an SQL query with optional alias
-    pub fn query_to_plan_with_alias(
+    /// Generate a logic plan from an SQL query.
```
Comment (alamb) [inline datafusion/sql/src/planner.rs:422]: This API makes a lot of sense -- thank you @jackwener

**Review by alamb — APPROVED** [review]
Very nice change @jackwener

**Pull request merged**